optashift-employee-rostering-shared/src/main/java/org/optaplanner/openshift/employeerostering/shared/shift/MovableShiftFilter.java

```diff
@@ -19,14 +19,13 @@
 import org.optaplanner.core.impl.heuristic.selector.common.decorator.SelectionFilter;
 import org.optaplanner.core.impl.score.director.ScoreDirector;
 import org.optaplanner.openshift.employeerostering.shared.roster.Roster;
-import org.optaplanner.openshift.employeerostering.shared.timeslot.TimeSlotState;
 
 public class MovableShiftFilter implements SelectionFilter<Roster, Shift> {
 
     @Override
     public boolean accept(ScoreDirector<Roster> scoreDirector, Shift shift) {
-        return !shift.isLockedByUser()
-                && shift.getTimeSlot().getTimeSlotState() != TimeSlotState.HISTORY;
+        // TODO: Check if shift is historic from RosterState
+        return !shift.isPinnedByUser();
```

Comment:
Once Shift.pinnedByUser has the @Pinned annotation, this boolean check isn't needed any more. But we 'll still have to check if it's not yet published (based on the published until date I guess, so the draftStartDate)

Reply:
https://issues.jboss.org/browse/PLANNER-1032